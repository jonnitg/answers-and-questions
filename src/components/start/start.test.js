import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Start from './start';

afterAll(cleanup);

it('renders score correctly', () => {
  const nextPhase = jest.fn();
  const { getByText } = render(
    <Start questionsAmount={5} nextPhase={nextPhase} />
  );
  const startTextElement = getByText(/questions/i);
  expect(startTextElement).toHaveTextContent(
    'This test has 5 questions to answer'
  );
});

it('clicks button', () => {
  const nextPhase = jest.fn();
  const { getByText } = render(
    <Start questionsAmount={5} nextPhase={nextPhase} />
  );
  const startButtonElement = getByText(/start/i);

  fireEvent.click(startButtonElement);
  expect(nextPhase.mock.calls.length).toEqual(1);
});

it('matches snapshot', () => {
  const nextPhase = jest.fn();
  const tree = renderer
    .create(<Start questionsAmount={5} nextPhase={nextPhase} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
