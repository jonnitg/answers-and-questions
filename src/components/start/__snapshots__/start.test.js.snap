// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-AxjAm iWCLJP"
>
  <h1
    className="sc-AxirZ fzdAyF"
  >
    Welcome!
  </h1>
  <p
    className="sc-AxiKw bOTXee"
  >
    This test has 
    5
     questions to answer
  </p>
  <button
    className="sc-AxhCb hJTwHZ"
    onClick={[MockFunction]}
    type="button"
  >
    Start
  </button>
</div>
`;
